### PR TITLE
feat: Integrate with Butter CMS

### PR DESCRIPTION
**Pull request by soniaklimas**
I want to merge this change because it integrates with ButterCMS.

# Pull Request Checklist

- [ ] Test the changes locally to ensure they work as expected.
- [ ] Document the testing process and results in the pull request description. (Screen recording, screenshot etc)
- [ ] Include new tests for any new functionality or significant changes.
- [ ] Ensure that tests cover edge cases and potential failure points.
- [ ] Document the impact of the changes on the system, including potential risks and benefits.
- [ ] Provide a rollback plan in case the changes introduce critical issues.
- [ ] Update documentation to reflect any changes in functionality.

### CODE REVIEW

**Review by karolkarolka — CHANGES_REQUESTED** [review]
I could seen some of the functions being moved from one dir to another like serializers, but I think that you've removed some code related to Saleor CMS and we should keep it

File: packages/infrastructure/src/public/saleor/cms-page/types.ts [hunk 1]
```diff
@@ -1,23 +1,3 @@
-import type { LanguageCodeEnum } from "@nimara/codegen/schema";
```
[the remark this answers is not in the document]
Comment (soniaklimas) [reply packages/infrastructure/src/public/saleor/cms-page/types.ts:1]: I just moved types related to use case to folder use-case (same as search case).

File: packages/infrastructure/src/public/saleor/cms-menu/infrastructure/cms-menu-get-infra.ts [hunk 2]
```diff
@@ -1,57 +1,9 @@
...
+import { serializeMenu } from "#root/lib/serializers/cms-menu";
+import type { CMSMenuGetInfra } from "#root/use-cases/cms-menu/types";
 
-import {
-  type MenuGet_menu_Menu_items_MenuItem,
-  MenuGetDocument,
-} from "../graphql/queries/generated";
-import type { CMSMenuGetInfra, SaleorCMSMenuServiceConfig } from "../types";
-
-const serializeMenuItem = ({
```
[the remark this answers is not in the document]
Comment (soniaklimas) [reply packages/infrastructure/src/public/saleor/cms-menu/infrastructure/cms-menu-get-infra.ts:11]: It is but I created separated file for serializers and move it there.

File: apps/storefront/src/lib/helpers.ts [hunk 3]
```diff
@@ -1,27 +1,6 @@
-import type { Attribute } from "@nimara/domain/objects/Attribute";
 import type { MenuItem } from "@nimara/domain/objects/Menu";
 import { loggingService } from "@nimara/infrastructure/logging/service";
 
-export const getAttributes = (
```
[the remark this answers is not in the document]
Comment (soniaklimas) [reply apps/storefront/src/lib/helpers.ts:5]: We do not use it anymore (there is new serializer which returns fields of cms data).

File: packages/infrastructure/src/public/saleor/cms-page/index.ts [hunk 4]
```diff
@@ -1,12 +0,0 @@
-import { cmsPageGetUseCase } from "#root/use-cases/cms-page/cms-page-get-use-case";
```
[the remark this answers is not in the document]
Comment (soniaklimas) [reply packages/infrastructure/src/public/saleor/cms-page/index.ts:1]: I just moved it to another file.

File: packages/infrastructure/src/lib/serializers/cms-menu.ts [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+          slug: item.name.toLowerCase(),
+          translation: null,
+        }
+      : null;
+
+    const pageSlug =
+      typeof item.page === "string"
+        ? item.page.split("/").filter(Boolean).pop() || ""
+        : "";
+    const page: Page | null = item.page
```
Comment (karolkarolka) [inline packages/infrastructure/src/lib/serializers/cms-menu.ts:68]:
```suggestion
    const page = item.page
```

**Review by karolkarolka — CHANGES_REQUESTED** [review]
one last very small change and you can merge

File: packages/infrastructure/src/lib/serializers/cms-menu.ts [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+      children: null,
+    };
+  });
+};
+
+export const serializeMenu = (
+  items: MenuGet_menu_Menu_items_MenuItem[] | ButterCMSMenuItem[],
+  source: "saleor" | "butterCms",
+): Menu => {
+  if (source === "saleor") {
```
Comment (wojtek007mirumee) [inline packages/infrastructure/src/lib/serializers/cms-menu.ts:94]: To jest problem trochę. Powinniśmy używać kompozycji zamiast budować takie warunki. 
Co myslisz o tym by stworzyć dwie funkcje: jedną dla saleor, drugą dla butter.
Wtedy jedna z nich moze byc uzyta w infrze buttera, a druga w infrze saleora.
Od poziomu infry powinnismy byc provider specifc, ale z wyraznym oddzieleniem kodu różnych providerów, by uniknąc tego ze zmiana dla jednego providera spowoduje bug dla drugiego.

File: packages/infrastructure/src/public/butter-cms/cms-menu/infrastructure/cms-menu-get-infra.ts [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+      return null;
+    }
+
+    const selectedMenu = menu.data.data.navigation_menu.find(
+      (menu: ButterCMSMenuItem) =>
+        menu.name.toLowerCase() === slug?.toLowerCase(),
+    );
+
+    return {
+      menu: serializeMenu(
```
Comment (wojtek007mirumee) [inline packages/infrastructure/src/public/butter-cms/cms-menu/infrastructure/cms-menu-get-infra.ts:32]: Tutaj i tak jesteśmy w kontekscie providera.
